features/feature-test/feature-test-suites/src/main/java/org/wso2/carbon/commons/ServiceAdminClient.java

```diff
@@ -0,0 +1,209 @@
+        try {
+            return serviceAdminStub.deleteFaultyServiceGroup(getFaultyData(serviceName).getArtifact());
+        } catch (RemoteException e) {
+            log.error("Faulty service deletion fails", e);
+            throw new RemoteException("Faulty service deletion fails", e);
+        }
+    }
+
+    /**
+     * Get all the services for a service name
+     *
+     * @param serviceName - service name to search
+     * @return ServiceMetaDataWrapper - service list
+     * @throws RemoteException - Error when list services.
+     */
+    public ServiceMetaDataWrapper listServices(String serviceName)
+            throws RemoteException {
+        ServiceMetaDataWrapper serviceMetaDataWrapper;
+        serviceMetaDataWrapper = serviceAdminStub.listServices("ALL", serviceName, 0);
+        serviceAdminStub.getFaultyServiceArchives(0);
+        return serviceMetaDataWrapper;
+    }
+
+    /**
+     * Get all the faulty services
+     *
+     * @return FaultyServicesWrapper - all faulty services.
+     * @throws RemoteException - Error when getting faulty service archives
+     */
+    public FaultyServicesWrapper listFaultyServices() throws RemoteException {
+        FaultyServicesWrapper faultyServicesWrapper;
+
+        faultyServicesWrapper = serviceAdminStub.getFaultyServiceArchives(0);
+
+        return faultyServicesWrapper;
+    }
+
+    /**
+     * Check whether service is available or not
+     *
+     * @param serviceName - service name
+     * @return boolean - service is available or not
+     * @throws RemoteException - Error when list services.
+     */
+    public boolean isServiceExists(String serviceName)
+            throws RemoteException {
+        boolean serviceState = false;
+        ServiceMetaDataWrapper serviceMetaDataWrapper;
+        ServiceMetaData[] serviceMetaDataList;
+        serviceMetaDataWrapper = listServices(serviceName);
+        serviceMetaDataList = serviceMetaDataWrapper.getServices();
+        if (serviceMetaDataList == null || serviceMetaDataList.length == 0) {
+            serviceState = false;
+        } else {
+            for (ServiceMetaData serviceData : serviceMetaDataList) {
+                if (serviceData != null && serviceData.getName().equalsIgnoreCase(serviceName)) {
+                    return true;
+                }
+            }
+        }
+        return serviceState;
+    }
+
+    /**
+     * Get service group by service name
+     *
+     * @param serviceName - service name
+     * @return - service group
+     * @throws RemoteException - Error when list services.
+     */
+    public String getServiceGroup(String serviceName) throws RemoteException {
+        ServiceMetaDataWrapper serviceMetaDataWrapper;
+        ServiceMetaData[] serviceMetaDataList;
+        serviceMetaDataWrapper = listServices(serviceName);
+        serviceMetaDataList = serviceMetaDataWrapper.getServices();
+        if (serviceMetaDataList != null && serviceMetaDataList.length > 0) {
```

Comment:
serviceMetaDataList != null && serviceMetaDataList.length > 0 both conditions are same no need to check both